src/components/Api.js

```diff
@@ -0,0 +1,115 @@
+import { routeData as routes } from "../utils/constants";
+const debug = false;
+// if(debug) console.log(``);
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  _buildRequest(routeData, body = undefined) {
+    const url = this._baseUrl + routeData.route;
+    const info = {
+      method: routeData.method,
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": this._contentType,
+      },
+    };
+
+    if (body) info.body = JSON.stringify(body);
+    if (debug) console.log(`call info:`);
+    if (debug) console.log(info);
+
+    return { url: url, info: info };
+  }
+
+  request(routeData, body) {
+    const requestInfo = this._buildRequest(routeData, body);
+    return fetch(requestInfo.url, requestInfo.info)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`API request Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err); // log the error to the console
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js` after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Reply:
Got it! Thank you; I was unsure where to include catches and where to not include them, and so I clearly was including them at random. Glad to have this rule of them.